Dockerfile

```diff
@@ -1,36 +1,68 @@
-FROM ruby:2.4.5-slim-stretch
+# XXX: using stretch here for pdftk dep, which is not availible after
+#      stretch (or in alpine) and is switched automatically to pdftk-java in buster
+#      https://github.com/department-of-veterans-affairs/va.gov-team/issues/3032
+
+###
+# shared configs for all child images, reuse these layers yo
+###
+FROM ruby:2.4.5-slim-stretch AS base
+
+ARG userid=993
+RUN groupadd -g $userid -r vets-api && \
+    useradd -u $userid -r -m -d /srv/vets-api -g vets-api vets-api
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+    dumb-init clamav imagemagick pdftk curl poppler-utils
+WORKDIR /srv/vets-api
+
+###
+# dev step; use --target=development to stop here
+# you will need to mount your source to /srv/vets-api to do anything useful here
+###
+FROM base AS development
 
 ARG sidekiq_license
-ENV BUNDLE_ENTERPRISE__CONTRIBSYS__COM=$sidekiq_license
 ARG exclude_sidekiq_ent
+ARG rails_env=development
+
+ENV BUNDLE_ENTERPRISE__CONTRIBSYS__COM=$sidekiq_license
 ENV EXCLUDE_SIDEKIQ_ENTERPRISE=$exclude_sidekiq_ent
-ENV APP_PATH /src/vets-api
-
-RUN groupadd -r vets-api && \
-useradd -r -g vets-api vets-api && \
-apt-get update -qq && \
-apt-get install -y build-essential \
-git \
-libpq-dev \
-libgmp-dev \
-clamav \
-imagemagick \
-pdftk \
-curl \
-poppler-utils && \
-freshclam
-
-WORKDIR $APP_PATH
-ADD . /src/vets-api
-
-RUN curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > /cc-test-reporter
-RUN chmod +x /cc-test-reporter
-
-COPY ./docker-entrypoint.sh /
-RUN chmod +x /docker-entrypoint.sh
-ENTRYPOINT ["/docker-entrypoint.sh"]
-
-ENV BUNDLE_PATH=/bundle \
-    BUNDLE_BIN=/bundle/bin \
-    GEM_HOME=/bundle
-ENV PATH="${BUNDLE_BIN}:${PATH}"
+ENV RAILS_ENV=$rails_env
+
+# only extra dev/build opts go here, common packages go in base 🖕
```

Comment:
Is this the finger you intended? 😂  I can't tell from the rendering on GitHub

Reply:
it was not, thanks!